h2/src/test/org/h2/test/unit/TestValue.java

```diff
@@ -285,8 +285,14 @@ private void testUUID() {
         ValueUuid min = ValueUuid.get(minHigh, minLow);
         assertEquals("00000000-0000-4000-8000-000000000000", min.getString());
 
-        ValueJavaObject vo = ValueJavaObject.getNoCopy(UUID.randomUUID(), null, null);
-        assertThrows(DbException.class, vo).convertTo(Value.UUID);
+        // Test conversion from ValueJavaObject to ValueUuid
+        ValueJavaObject valObj = ValueJavaObject.getNoCopy(UUID.fromString("12345678-1234-4321-8765-123456789012"), null, null);
+        Value valUUID = valObj.convertTo(Value.UUID);
+        assertTrue(valUUID instanceof ValueUuid);
+        assertTrue((valUUID.getString().equals("12345678-1234-4321-8765-123456789012")));
```

Comment:
Here it would be a good idea to add assertion that `valUUID.getObject().equals(originalUUID)`